### DISCUSSION TITLE
How to upload image files using GraphQL plugin?

### DISCUSSION THREAD

**Discussion by kaanguru** [strapi/strapi#6196] [Legacy Topics]
Hi
**Do I need to convert my image to a Base64 string?**
It seems the most popular frontends uses a **vue-apollo** but there is no tutorial with image upload support.
People will think that it's not possible to upload images to strapi server. All restaurant tutorials are just using URL input areas.

Check out the detailed question.
https://stackoverflow.com/q/61733391/11993949

**NOTE**:  It's hard to guess what kind of data does strapi wants from [upload files docs](https://strapi.io/documentation/3.0.0-beta.x/plugins/upload.html#upload-files)

**Comment by lauriejim** [comment]
Hello! You will find 2 gist there that talk about Upload files w/ GraphQL

https://gist.github.com/alexandrebodin

**Comment by hihurmuz** [comment]
did anyone try upload image with aws s3 ?

**Reply by lauriejim** [reply]
Here it is to install AWS S3 provider - https://strapi.io/documentation/v3.x/plugins/upload.html#using-a-provider
